Add mentions that using target directives are experimental (#2262)
VirtusLab/scala-cli: website/docs/commands/test.md

A source file is treated as test source if:

- the file name ends with `.test.scala`, or
- the file comes from a directory that is provided as input, and the relative path from that file to its original
  directory contains a `test` directory
- it contains the `//> using target.scope test` directive, or

:::caution

> The `using target` directives are an experimental feature. They may be subject to changes in the future.

Reviewer: ```suggestion
The `using target` directives are an experimental feature, and may change in future versions of Scala CLI.
```
